pkg/operator/mirror.go

```diff
@@ -422,10 +422,12 @@ func (o *MirrorOptions) associateDeclarativeConfigImageLayers(mappingDir string,
 
 		srcDir := filepath.Join(o.RootDestDir, config.SourceDir)
 		assocs, err := image.AssociateImageLayers(srcDir, imgMappings, images)
-		if merr, ok := err.(*image.MirrorError); ok {
-			logrus.Warn(merr)
-		} else {
-			return err
+		if err != nil {
+			if merr, ok := err.(*image.MirrorError); ok {
+				logrus.Warn(merr)
+			} else {
+				return err
+			}
```

Comment:
Casting this error may work sometimes, but not if the error has a parent. Use [`errors.As`](https://pkg.go.dev/errors?utm_source=godoc#example-As) here instead:
```suggestion
			merr := &image.MirrorError{}
			if !errors.As(err, &merr) {
				return err
			}
			logrus.Warn(err)
```